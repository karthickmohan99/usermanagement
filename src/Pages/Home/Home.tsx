/* eslint-disable @typescript-eslint/no-explicit-any */
import React, { useEffect, useState } from 'react';
import { Table, Button, Navbar, Nav } from 'react-bootstrap';
import { useDispatch, useSelector } from 'react-redux';
import { getuser } from '../../State /Action/Action';
import { useNavigate } from 'react-router';
import Popup from '../../Components/Popup';

export interface EditData {
  email: string;
  id: number | string;
  name: string;
  password: string;
  i: number;
}
const Home = () => {
  const dispatch: any = useDispatch();

  const navigate = useNavigate();
  const user = useSelector((state: any) => state.allreducer.user);

  console.log(user, 'Homeuseselector');
  const [userDatas, setUserDatas] = useState<string | number>('0');
  const [showModal, setShow] = useState<boolean>(false);

  const handleClose = () => setShow(false);

  //delete onClick
  const onDelete = (e: React.MouseEvent<HTMLElement>, userData: string | number): void => {
    e.preventDefault();
    setShow(true);
    setUserDatas(userData);
  };

  //edit onClick
  const handleEdit = (userData: EditData) => {
    console.log(userData, 'type of userdata');
    navigate(`/Modals/${userData.id}`);
  };

  //add onClick
  const onAdd = () => {
    navigate('/add');
  };

  useEffect(() => {
    dispatch(getuser());
    setTimeout(() => {
      dispatch(getuser());
    }, 60000);
  }, [dispatch, navigate]);

  return (
    <>
      <Navbar bg="dark" variant="dark">
        <Navbar.Brand href="#home">HOME</Navbar.Brand>
        <Nav className="me-auto">
          <Nav.Link href="/">Logout</Nav.Link>
        </Nav>
      </Navbar>
      <div>
        <h1 style={{ textAlign: 'center' }}> welcome Home </h1>

        <Table striped bordered hover data-testid="table">
          <thead>
            <tr>
              <th>Sl.No</th>
              <th>user ID</th>
              <th>name</th>
              <th>Email</th>
            </tr>
          </thead>
          <tbody>
            {user.length > 0 &&
              user.map((userData: EditData, i: number) => {
                return (
                  <tr key={userData.i}>
                    <td>{i + 1}</td>
                    <td>{userData.id}</td>
                    <td>{userData.name}</td>
                    <td>{userData.email}</td>
                    <td>
                      <Button
                        onClick={() => {
                          handleEdit(userData);
                        }}
                        variant="dark"
                        data-testid="editid"
                        id="editbutton"
                        className="classedit">
                        Edit
                      </Button>
                    </td>

                    <td>
                      <Button
                        onClick={(e) => {
                          onDelete(e, userData.id);
                        }}
                        variant="danger"
                        data-testid="deleteFn"
                        id="deletebutton"
                        className="classdelete">
                        Delete
                      </Button>
                    </td>
                  </tr>
                );
              })}
          </tbody>
        </Table>
        <Button onClick={onAdd} className="addButton" variant="info">
          Add user
        </Button>
        {showModal && <Popup shows={showModal} onHides={handleClose} datas={userDatas} />}
      </div>
    </>
  );
};
export default Home;
