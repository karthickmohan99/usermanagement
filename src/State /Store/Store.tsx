import reducers from '../Reducer/Combinedreducer';
import ReduxThunk from 'redux-thunk';
import { createStore, applyMiddleware } from 'redux';

export const Store = createStore(reducers, applyMiddleware(ReduxThunk));

export type AppDispatch = typeof Store.dispatch;
