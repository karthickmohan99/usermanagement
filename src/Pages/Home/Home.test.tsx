/* eslint-disable @typescript-eslint/no-explicit-any */
import { render as rtlrender, cleanup, screen, fireEvent } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter as Router } from 'react-router-dom';
import { Store } from '../../State /Store/Store';
import userEvent from '@testing-library/user-event';
import Home from './Home';

describe('Home component', () => {
  const render = (component: any) => rtlrender(<Provider store={Store}>{component}</Provider>);
  afterEach(cleanup);
  it('should check the  component renders properly', () => {
    //arrange
    render(
      <Router>
        <Home />
      </Router>
    );

    const titleValue = screen.getByText('welcome Home');
    expect(titleValue).toHaveTextContent('welcome Home');
  });

  it('should renders the table', () => {
    render(
      <Router>
        <Home />
      </Router>
    );

    const table = screen.getByTestId('table');
    expect(table).toBeInTheDocument();
  });

  it('checks the edit button', () => {
    render(
      <Router>
        <Home />
      </Router>
    );

    // eslint-disable-next-line testing-library/no-node-access
    const editbutton = document.querySelector('.classedit');
    expect(editbutton).toBeDefined();
  });
  it('checks the  delete button', () => {
    render(
      <Router>
        <Home />
      </Router>
    );
    // eslint-disable-next-line testing-library/no-node-access
    const Deletebutton = document.querySelector('.classdelete');
    expect(Deletebutton).toBeDefined();
  });
  it('checks the  delete button', () => {
    render(
      <Router>
        <Home />
      </Router>
    );
    // eslint-disable-next-line testing-library/no-node-access
    const Deletebutton = document.querySelector('.classdelete');
    expect(Deletebutton).toBeDefined();
  });

  // it('checks the  delete button', () => {
  //   render(
  //     <Router>
  //       <Home />
  //     </Router>
  //   );
  //   // eslint-disable-next-line testing-library/no-node-access
  //   const Deletebutton = document.querySelector('.classdelete');
  //   expect(Deletebutton).toBeDefined();
  //   const dbutton = jest.fn();
  //   fireEvent.click(Deletebutton);
  //   expect(dbutton).toHaveBeenCalledTimes(0);
  // });
});
