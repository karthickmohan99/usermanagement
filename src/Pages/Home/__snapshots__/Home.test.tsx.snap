// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the button should match with snapshot 1`] = `[Function]`;
